oat-sa/tao-core-libs-fe
# Feature/TR-4862/Unify handlebars helpers

### Related to: https://oat-sa.atlassian.net/browse/TR-4862

### Summary

Refactor the Handlebars integration in order to only have one place where it is made:
- move the helpers to a separate factory in order to share and include them from a common place.
- extract the module writer to a separate module in order to allow replacing it when needed.
- make the Handlebars integration shareable.

### Details

The `tpl.js` file is responsible for the integration of Handlebars with RequireJS, together with the declaration of the custom helpers for TAO. However, it was duplicated in several places. The purpose of this PR is to have only one place where this integration is made, and allows to replace some of its pieces.

In some places, only the helpers were duplicated, hence the idea of separating them into a dedicated module.
The `tpl.js` contains the RequireJS plugin for compiling the templates on the fly, or adding them into a bundle. With this respect, it generates wrapping bundles containing the compiled template, together with a reference to Handlebars. However, in some cases, it is needed to link the template to a specific version of Handlebars (PCI case). For this reason, the module writer has been split into a dedicated module, which can be replaced using the RequireJS config.

Unit tests have been moved too.

### How to test
Install all dependencies, then
- check the Handlebars template still behave well in debug mode
- check the Handlebars template still behave well in production mode (using the bundles)
- run the unit tests: `npm test`
